ios/brave-ios/Sources/BraveShields/BraveShieldsSettings+Extensions.swift

```diff
@@ -0,0 +1,157 @@
+        let urlToMigrate = URL(string: urlString)
+      {
+        let fingerPrintingProtection: Bool =
+          fingerprintingProtectionDomainToMigrate.isShieldExpected(
+            .fpProtection,
+            considerAllShieldsOption: false
+          )
+        setFingerprintMode(fingerPrintingProtection ? .standardMode : .allowMode, for: urlToMigrate)
+      }
+
+      // Block Scripts
+      let domainsWithExplicitBlockScripts = domains.filter { $0.shield_noScript != nil }
+      let blockScriptsDomainToMigrate = domainsWithExplicitBlockScripts.sortedForMigration().first
+      // only assign if an explicit value assigned, else default is used
+      if let blockScriptsDomainToMigrate,
+        let urlString = blockScriptsDomainToMigrate.url,
+        let urlToMigrate = URL(string: urlString)
+      {
+        let blockScripts: Bool = blockScriptsDomainToMigrate.isShieldExpected(
+          .noScript,
+          considerAllShieldsOption: false
+        )
+        setBlockScriptsEnabled(blockScripts, for: urlToMigrate)
+      }
+    }
+
+    // Second, migrate shield settings that use domain pattern.
+    // Auto-Shred level uses domain pattern to store content settings.
+    // This differs from `Domain` which includes the scheme in the key
+    // (`domainURL`). As such, we may have 2 `Domain` models that point
+    // to 1 domain in content settings, for example
+    // https://brave.com & https://accounts.brave.com.
+    let groupedByDomain: [String: [Domain]] = Dictionary(grouping: domains) { domain in
+      guard let urlString = domain.url, let url = URL(string: urlString) else { return "" }
+      return domainPatternFromURL(url)
+    }
+    for (domain, domains) in groupedByDomain {
+      // if no url, domain will be an empty string. Shouldn't occur so we can ignore.
+      guard !domain.isEmpty else { continue }
+
+      let domainsWithExplicitShredLevel = domains.filter { $0.shield_shredLevel != nil }
+      let shredLevelDomainToMigrate = domainsWithExplicitShredLevel.sortedForMigration().first
+      // only assign if an explicit value assigned, else default is used
+      if let shredLevelDomainToMigrate,
+        let urlString = shredLevelDomainToMigrate.url,
+        let urlToMigrate = URL(string: urlString)
+      {
+        let autoShredMode = shredLevelDomainToMigrate.shredLevel.autoShredMode
+        self.setAutoShredMode(autoShredMode, for: urlToMigrate)
+      }
+    }
+  }
+
+  private func hostPatternFromURL(_ url: URL) -> String {
+    return (url as NSURL).hostPatternString
+  }
+
+  private func domainPatternFromURL(_ url: URL) -> String {
+    return (url as NSURL).domainPatternString
+  }
+}
+
+extension Array where Element == Domain {
+  fileprivate func sortedForMigration() -> [Domain] {
+    sorted(by: { lhs, rhs in
+      let isLHSHttps = lhs.url?.hasPrefix("https://") == true
+      let isRHSHttps = rhs.url?.hasPrefix("https://") == true
```

Comment:
Is the domain string always normalized to lowercase?

Reply:
Not guaranteed,  good catch. 
It seems unlikely to not be a lowercase scheme, as the `URL` passed to `Domain.getOrCreate` will have lowercase scheme even when an uppercase scheme is entered into the omni bar, but I'll update this to be sure.